File: WebMonitor/ClientApp/src/model/nvidia-refresh-setting.ts
export enum NvidiaRefreshSetting {
    Enabled = 0,
    PartiallyDisabled = 1,
    Disabled = 2,
    LongerInterval = 3
}

export interface NvidiaRefreshSettings {
    refreshSetting: NvidiaRefreshSetting;
    nRefreshIntervals: number;
}

File: WebMonitor/Model/NvidiaRefreshSetting.cs
namespace WebMonitor.Model;

/// <summary>
/// Settings for how Nvidia GPU monitoring should be handled
/// </summary>
public enum NvidiaRefreshSetting
{
    /// <summary>Everything is enabled</summary>
    Enabled,
    /// <summary>Clock and power are disabled</summary>
    PartiallyDisabled,
    /// <summary>Everything is disabled</summary>
    Disabled,
    /// <summary>Everything is enabled but refreshes occur less often</summary>
    LongerInterval
}

/// <summary>
/// Interface for settings for how Nvidia GPU monitoring should be handled
/// </summary>
public class NvidiaRefreshSettings
{
    /// <summary>
    /// Refresh setting
    /// </summary>
    public NvidiaRefreshSetting RefreshSetting { get; internal set; }

    /// <summary>
    /// Number of refresh intervals to wait between refreshes
    /// </summary>
    public int NRefreshIntervals { get; internal set; }
}
